connman/gdbus/client.c

```diff
@@ -853,28 +853,30 @@ gboolean g_dbus_proxy_method_call(GDBusProxy *proxy, const char *method,
 	if (client == NULL)
 		return FALSE;
 
-	data = g_try_new0(struct method_call_data, 1);
-	if (data == NULL)
-		return FALSE;
-
-	data->function = function;
-	data->user_data = user_data;
-	data->destroy = destroy;
-
 	msg = dbus_message_new_method_call(client->service_name,
 				proxy->obj_path, proxy->interface, method);
-	if (msg == NULL) {
-		g_free(data);
+	if (msg == NULL)
 		return FALSE;
-	}
 
 	if (setup) {
 		DBusMessageIter iter;
 
 		dbus_message_iter_init_append(msg, &iter);
-		setup(&iter, data->user_data);
+		setup(&iter, user_data);
 	}
 
+	if (!function)
+		return g_dbus_send_message(client->dbus_conn, msg);
+
+	data = g_try_new0(struct method_call_data, 1);
+	if (data == NULL)
+		return FALSE;
```

Comment:
after being moved, this mid-function return now leaks msg

Reply:
Anyways this kind of small allocations should take the whole ConnMan down according to [updated coding style](https://git.kernel.org/pub/scm/network/connman/connman.git/tree/doc/coding-style.txt#n158), so, well, perhaps we could add something here.

Reply:
But if you really think about this. If a small memory allocation fails it reeks of a much bigger issue as memory is running out. Do you think the software will continue run after this? I think not because then there are gazillion other small reservations that start to fail and this just ends up dying at some other point. So memory leak in such case has no significance whatsoever. I personally do not see any use of putting any effort towards this.

The struct itself is not a really bug one, two function pointers and a void pointer. So according to the coding style this should be left to crash anyways.

Reply:
Well, it is at least in sync with the rest of the code after [3b0a203](https://github.com/sailfishos/connman/pull/66/commits/3b0a2036ecfa0647b90cf602ad19319a1c907d17)